supercommand.go

```diff
@@ -483,17 +493,30 @@ func (c *SuperCommand) Init(args []string) error {
 	} else {
 		subcmd.SetFlags(c.commonflags)
 	}
-	if err := c.commonflags.Parse(subcmd.AllowInterspersedFlags(), args); err != nil {
+	if err := c.commonflags.Parse(subcmd.AllowInterspersedFlags(), cleanArgs); err != nil {
 		return err
 	}
 
-	args = c.commonflags.Args()
+	cleanArgs = c.commonflags.Args()
 	if c.showHelp {
 		// We want to treat help for the command the same way we would if we went "help foo".
-		args = []string{c.action.name}
+		cleanArgs = []string{c.action.name}
 		c.action = c.subcmds["help"]
 	}
-	return c.action.command.Init(args)
+
+	err := c.action.command.Init(cleanArgs)
+
+	// Commands may intentionally return a command missing
+	// error during init to trigger their missing callback.
+	if stderr.Is(err, ErrCommandMissing) {
```

Comment:
This would be cleaner as
```
if !errors.Is(err, errCommandMissing) || c.missingCallback == nil {
  return err
}
setupMissingCallback()
return nil
```